Kconfig new libc changes cause echo server cmake error

After libc config changes

9555f82d28
0e59f6846f
c1f865b412

echo server cmake fails:

```
cd samples/net/sockets/echo_server/build
cmake  -DBOARD=native_posix -DCONF_FILE=prj.conf  ..
```

Results in cmake error:
```
warning: the choice symbol NEWLIB_LIBC (defined at lib/libc/Kconfig:29) is selected by the following symbols, but select/imply has no effect on choice symbols
 - MIPI_SYST_LIB (defined at modules/Kconfig.syst:4)

Error: Aborting due to non-whitelisted Kconfig warning 'warning: the choice symbol NEWLIB_LIBC
(defined at lib/libc/Kconfig:29) is selected by the following symbols, but select/imply has no
effect on choice symbols  - MIPI_SYST_LIB (defined at modules/Kconfig.syst:4)'. Note: If this
warning doesn't point to an actual problem, you can add it to the whitelist at the top of
zephyrproject/zephyr/scripts/kconfig/kconfig.py.

CMake Error at zephyrproject/zephyr/cmake/kconfig.cmake:210 (message):
  command failed with return code: 1
Call Stack (most recent call first):
  zephyrproject/zephyr/cmake/app/boilerplate.cmake:461 (include)
  CMakeLists.txt:5 (include)


```

--- a/modules/Kconfig.syst
+++ b/modules/Kconfig.syst
@@ -3,6 +3,6 @@
 
 config MIPI_SYST_LIB
 	bool "MIPI SyS-T Library Support"
-	select NEWLIB_LIBC
+	select REQUIRES_FULL_LIBC
 	help
 	  This option enables the MIPI SyS-T Library

